lib/adminware/config.rb

```diff
@@ -0,0 +1,64 @@
+require 'yaml'
+require 'fileutils'
+
+module Adminware
+  class Config
+    #Set default values for config settings
+    DEFAULT_JOBDIR = "jobs/"
+    DEFAULT_LOGFILE = "logs/adminware.log"
+    DEFAULT_STATEDIR = "var/state.yaml"
+
+    #Load the config file
+    def initialize
+      @path = Adminware.root 
+      configfile = File.join(@path, 'etc/config.yaml')
+      @config = YAML.load_file(configfile)
+
+      #If no path given in config then set them to their defaults
+      @jobdir = @config['centraljobdir'] ||= DEFAULT_JOBDIR
+      @jobdir = set_path(@jobdir)
+
+      @logfile = @config['logfile'] ||= DEFAULT_LOGFILE
+      @logfile = set_path(@logfile)
+
+      @statedir = @config['statedir'] ||= DEFAULT_STATEDIR
+      @statedir = set_path(@statedir)
+    end
+
+    #These methods return their respective files
+    def jobdir
+      @jobdir
```

Comment:
You could relocate the work done in the constructor here instead, which would be a little more Ruby-ish, i.e. don't set `@jobdir` in the constructor, but instead:

```
def jobdir
  @jobdir ||= resolve_path(@config['centraljobdir'] || DEFAULT_JOBDIR)
end
```